### PR TITLE
Add test to make sure suffix sharding prevents prefix clustering.

### LINKED ISSUE #8: Add a test to make sure suffix sharding prevents prefix clustering
Lets make sure to add a test that checks that adding a bunch of keys with a shared prefix doesnt result in all the entries landing in the same directory with suffixes chosen.

### PR DESCRIPTION
**Pull request by kevina**
Closes #8

### CODE REVIEW

**Comment by kevina** [comment]
Note: based on top of #11.  Can rebase if required once #11 is merged.

File: flatfs_test.go [hunk 1]
```diff
@@ -400,6 +401,50 @@ func testBatchDelete(dirFunc mkShardFunc, t *testing.T) {
...
+	tempdir, cleanup := tempdir(t)
+	defer cleanup()
+
+	fs, err := flatfs.New(tempdir, flatfs.NextToLast(1), false)
+	if err != nil {
+		t.Fatalf("New fail: %v\n", err)
+	}
+
+	r := rand.New()
+	N := 1024 // should be divisible by 32 so the math works out
```
Comment (Kubuxu) [inline flatfs_test.go]: I would make it in for of `32*X` so math always works out.

File: flatfs_test.go [hunk 2]
```diff
@@ -400,6 +401,50 @@ func testBatchDelete(dirFunc mkShardFunc, t *testing.T) {
...
+
+	dirs, err := ioutil.ReadDir(tempdir)
+	if err != nil {
+		t.Fatalf("ReadDir fail: %v\n", err)
+	}
+	if len(dirs) != 32 {
+		t.Fatalf("Expected 32 directories in %s", tempdir)
+	}
+	idealFilesPerDir := float64(N) / 32.0
+	tolerance := math.Floor(idealFilesPerDir * 0.50)
```
Comment (Kubuxu) [inline flatfs_test.go]: That is 50% tolerance, right? Can we still reliably get closer than this?

We might also make the test bigger but make the seed static, as currently there is remote but non-zero chance of this test failing if stars align correctly.

File: flatfs_test.go (hunk 1 above)
Comment (kevina) [inline flatfs_test.go]: I prefer a number a direct number.  If we use 32* I will have to stop and do the math to figure the numbers of key used.  The comment also makes it explicit.  With just a 32*X then I would wonder why it is written like that.  @whyrusleeping what's your view.

File: flatfs_test.go (hunk 2 above)
Comment (kevina) [inline flatfs_test.go]: Okay I was able to decrease it to 20%.
